pkg/controller/clone-controller.go

```diff
@@ -179,7 +179,7 @@ func (r *CloneReconciler) Reconcile(_ context.Context, req reconcile.Request) (r
 	}
 	log := r.log.WithValues("PVC", req.NamespacedName)
 	log.V(1).Info("reconciling Clone PVCs")
-	if pvc.DeletionTimestamp != nil || !r.shouldReconcile(pvc, log) {
+	if pvc.DeletionTimestamp != nil || (!r.shouldReconcile(pvc, log) && !isPodRetainAfterCompletion(pvc)) {
```

Comment:
We do want to remove the finalizer from the PVC when the clone is done, just not remove the pod if the annotation is set. So I would modify r.cleanup to not delete pod instead of skipping the entire function (which also removes the finalizer)

Reply:
You mean when user manually deletes the pod but the PVC still exists, it will still have the finalizer, which is risky/undesirable?

Reply:
Its undesirable because now for some reason unknown to the user, they cannot delete the PVC anymore (due to the finalizer)

Reply:
Only if we already deleted the pod, right?

Reply:
finalizer on PVC should be removed regardless of this annotation

Reply:
done